### PR TITLE
Refactor HD Wallets for Enhanced Security

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Refactor

**What does this PR do? Why is it needed?**

Our current HD wallet is extremely insecure, and other teams do not follow our convention. Currently:
- We create a mnemonic for a user and tell them to write it down
- We encrypt the seed derived from the mnemonic with their wallet password
- We store the encrypted seed on disk
- We load the seed to RAM upon startup of the validator
- We generate N keys as needed at runtime by deriving them from the seed

If someone takes this seed, or dumps a validator's memory, they are screwed. Moreover, storing a seed on disk is still not an adequate substitute.

This PR advocates for a different approach to our HD wallet altogether. Instead, we make it a **simple wrapper** around our imported keymanager. During wallet creation, we now show the user a mnemonic and ask them to write it offline, then we ask the user to generate N accounts where N > 0. Then, we import those as keystore files into our imported keymanager. The seed nor mnemonic are never stored after this.

This PR also removes the `accounts create` command the RPC endpoint given there is no longer any way to create a new account in Prysm. This must be done by another tool such as ethdo or the deposit cli.

### CODE REVIEW

File: validator/accounts/wallet_create.go [hunk 1]
```diff
@@ -9,7 +9,6 @@ import (
 	"github.com/manifoldco/promptui"
```
Comment (rauljordan) [inline validator/accounts/wallet_create.go:9]: Review these changes

File: validator/accounts/wallet_recover.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ type RecoverWalletConfig struct {
 	WalletDir        string
```
Comment (rauljordan) [inline validator/accounts/wallet_recover.go:36]: Review these changes

File: validator/keymanager/derived/keymanager.go [hunk 3]
```diff
@@ -6,61 +6,30 @@ import (
 	"fmt"
```
Comment (rauljordan) [inline validator/keymanager/derived/keymanager.go:6]: Review this file

File: validator/accounts/wallet_recover.go [hunk 4]
```diff
@@ -138,39 +137,21 @@ func RecoverWallet(ctx context.Context, cfg *RecoverWalletConfig) (*wallet.Walle
...
-			return nil, nil, errors.Wrap(err, "could not create account in wallet")
-		}
-		depositDataList[i] = depositData
-	}
 	log.WithField("wallet-path", w.AccountsDir()).Infof(
 		"Successfully recovered HD wallet with %d accounts. Please use accounts list to view details for your accounts",
 		cfg.NumAccounts,
 	)
-	return w, depositDataList, nil
+	return w, nil, nil
```
Comment (prestonvanloon) [inline validator/accounts/wallet_recover.go:154]: 2nd param is always nil. Want to refactor the method signature?

File: validator/keymanager/derived/mnemonic.go [hunk 5]
```diff
@@ -4,29 +4,43 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/pkg/errors"
+
 	"github.com/prysmaticlabs/prysm/shared/promptutil"
+	"github.com/prysmaticlabs/prysm/shared/rand"
+
 	"github.com/tyler-smith/go-bip39"
```
Comment (prestonvanloon) [inline validator/keymanager/derived/mnemonic.go:12]: Please fix

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**